www/docs/client/mocking.md

```diff
@@ -0,0 +1,234 @@
+---
+id: Mocking
+title: Mocking your server
+sidebar_label: Mocking your server
+slug: /mocking
+---
+
+Mocking tRPC gives you the ability to isolate your frontend components into testing or presentation environments without having to alter how they behave.
+
+By intercepting our tRPC calls at the [Service Worker level](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) and returning responses as the component expects, mocking can help you to:
+
+- Write performant, easy to execute tests that don't rely on changing the underlying component in any way
+- Present componenets in frontend workshops like Storybook exactly as they would appear and behave in your main app
+
+The examples below use [Jest](https://facebook.github.io/jest/docs/en/tutorial-react.html), (React Testing Library)[https://github.com/testing-library/react-testing-library], and [Mock Service Worker](https://www.npmjs.com/package/msw-trpc) (MSW), but the concepts apply to any testing framework.
+
+## Creating a TRPCMockedProvider
+
+Every test for a React component that uses your `tRPC client` must make it available in React's context.
+
+- In React, you achieve this by wrapping your component tree with the `trpc.Provider` and `QueryClientProvider` components.
+- In Next.js, this step is abstracted in `_app.ts` by using `trpc.withTRPC(MyApp)`.
+
+In tests however, this context is not avaliable by default. Instead, you will need to expose this context by creating a `TRPCMockedProvider`.
```

Comment:
This section reads a bit confusing to me. Here's my attempt:

In order for tRPC's hooks to work, they require context that describes the URL of the backend and the structure of the API.

- In React, you achieve this by wrapping your component tree with the `trpc.Provider` and `QueryClientProvider` components.
- In Next.js, this step is abstracted in `_app.ts` by using `trpc.withTRPC(MyApp)`.

We still need this context when using our components in a testing environment, but we want the requests to be made to a mocked version of our API as this allows us test the frontend in isolation. For this, we will need a mocked provider.

Reply:
Thank you. I like your version a lot better.

I've made two minor, additional tweaks:
- I've added a link to our documentation on context, as I think that if you don't have context on what `context` is, this sentence could be confusing. 
- I've played around the the phrasing of the last paragraph to make it clear that our mocked provider is enabling the component to make requests as it normally would and that mocking the API is something we'll do later. It's minor change but I think this illustrates the difference that this current step is making a little more clearly

> In order for tRPC's hooks to work, they require [context](https://trpc.io/docs/context) that describes the URL of the backend and the structure of the API.

> - In React, you achieve this by wrapping your component tree with the `trpc.Provider` and `QueryClientProvider` components.
> - In Next.js, this step is abstracted in `_app.ts` by using `trpc.withTRPC(MyApp)`.

> We still need this context when using our components in an isolated environment, because they still need to be capable of making requests that we will later intercept and pass through the mocked version of our API. For this, we will need to create and use a mocked provider.